src/sentry/integrations/discord/requests/base.py

```diff
@@ -0,0 +1,107 @@
+from __future__ import annotations
+
+import dataclasses
+from collections.abc import Mapping
+
+from rest_framework import status
+from rest_framework.request import Request
+
+from sentry import options
+from sentry.services.hybrid_cloud.integration import RpcIntegration, integration_service
+
+from ..utils import logger, verify_signature
+
+
+@dataclasses.dataclass(frozen=True)
+class DiscordRequestError(Exception):
+    """
+    Something was invalid about the request from Discord.
+    Includes the status the endpoint should return, based on the error.
+    """
+
+    status: int
+
+
+class DiscordRequest:
+    """
+    A Request from Discord to our interactions endpoint.
+
+    Handles request verification and data access.
+
+    Raises DiscordRequestError whenever something goes wrong, including the
+    appropriate response code that the endpoint should respond with.
+    """
+
+    def __init__(self, request: Request):
+        self.request = request
+        self._integration: RpcIntegration | None = None
+        self._data: Mapping[str, object] = {}
+
+    @property
+    def integration(self) -> RpcIntegration | None:
+        return self._integration
+
+    @property
+    def data(self) -> Mapping[str, object]:
+        if not self._data:
+            self._validate_data()
+        return self._data
+
+    @property
+    def guild_id(self) -> str:
+        return str(self.data.get("guild_id", None))
+
+    @property
+    def logging_data(self) -> Mapping[str, str | int]:
+        _data: Mapping[str, object] = self.request.data
+        # TODO: come back to this later and see what additional metadata makes sense to include here
+        data: dict[str, str | int | None] = {
+            "discord_guild_id": str(_data.get("guild_id", None)),
```

Comment:
I think `str(None)` evaluates to `'None'`. Is that what we want here?

Reply:
Good catch. Slack does `None` not `'None'`, so I'll go with that.